server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -733,6 +738,35 @@ private boolean updateSettings(Settings toApply, Settings.Builder target, Settin
         return changed;
     }
 
+    private void validate(Settings toApply, Settings target) {
+        Settings toValidate = copyValidSettings(target).put(toApply.filter(NOT_WILDCARD_SETTING)).build();
+        validate(toValidate, false);
+    }
+
+    private Settings.Builder copyValidSettings(Settings settings) {
+        Settings.Builder validSettings = Settings.builder();
+        settings.keySet().stream().filter(NOT_WILDCARD_SETTING).forEach(key -> {
+            try {
+                validate(key, settings, false);
+                validSettings.copy(key, settings);
+            } catch (Exception ignored) {
+                // ignore invalid settings
+            }
+        });
+        return validSettings;
+    }
+
+    private boolean hasChanged(Settings toApply, Settings.Builder target) {
+        boolean changed = toApply.keySet().stream().anyMatch(k -> k.endsWith("*"));
+        if (!changed) {
```

Comment:
The house style is to avoid using the `!` unary operator and to prefer saying `changed == false` here, because `!` is small and easy to miss in a long expression, and `== false` is larger and therefore more visible. I think this code will go away due to other comments, but thought I'd raise it anyway for your info.